front/app/containers/Admin/projects/project/index.test.tsx

```diff
@@ -150,29 +164,33 @@ describe.skip('<AdminProjectEdition />', () => {
     );
     const routerProps = getRouterProps('continuousTypeform');
 
-    // passing in mock data to replace the HoCs
-    // ignore the error, shallowWithIntl method passes in the intl object
-    const wrapper = shallowWithIntl(
+    render(
       <AdminProjectsProjectIndex
         surveys_enabled={surveys_enabled}
         typeform_enabled={typeform_enabled}
         phases={phases}
-        granularPermissionsEnabled={granularPermissionsEnabled}
         project={project}
-        children={children}
+        intl={intl}
         {...routerProps}
         {...localizeProps}
-      />
+        {...additionalProps}
+      >
+        {children}
+      </AdminProjectsProjectIndex>
     );
-    const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;
 
-    expect(tabs.map((tab) => tab.url)).toMatchSnapshot();
+    const tabNames = ['General', 'Description', 'Events'];
```

Comment:
Wondering whether we can load these with formatMessage?

Reply:
I think if we're using FormatMessage to generate both the component's copy and the message we're testing against then a bug in FormatMessage (however unlikely) would't cause the tests to fail. Explicit strings for the test tab names means we know exactly what the component should look like when rendered with these props. If the default tab names change it's easy to update them here, and if the test breaks because one of the explicit tabs is no longer present then it means we caught a bug

Reply:
Yeah. I was just thinking that it's easier to update if you have formatMessage here. 

If you change a component, you wouldn't search for this copy and forget about updating tests. CI will remind you, but it's an extra fix, running CI again, waiting, etc. I think that's more value than the indeed unlikely event of a formatMessage bug. :-) 

But I'm a testing newbie, so I don't know what's the best practice here.

Reply:
I don't think there's necessarily a right answer, testing is notoriously difficult to have actual best practices for since everyone's idea about what should be tested and how is different.

My take on it is that using FormatMessage in the test file makes the test more loose, the test checking that strings match will pass if FormatMessage breaks, or if the default message somehow has a typo in it, in each case there would be the same output in both places and so the test would pass. By hardcoding the strings in the test file the test will break if the displayed string changes for any reason in the actual component, which is either on purpose (so manually confirm the change is meant to be and update the test file with the output from the failed test's diff) or not on purpose, which means we caught a bug or error.

The argument against that is that the tests should be more flexible and that we aren't testing FormatMessage here, just that the tabs show whatever is passed into them, so making it a manual task to update the tests if the copy is changed on purpose in the future creates more work down the road. 

As far as running CI again, I've found that running tests in watch mode while developing a component (especially one with existing tests) is a pretty instantaneous feedback loop for the state of the unit tests, and helps me catch things that I've broken much earlier in the dev cycle.